keps/sig-architecture/20190731-production-readiness-review-process.md

```diff
@@ -0,0 +1,258 @@
+
+* TBD: Document some experiences that motivate the need for this.
+
+### Goals
+
+* Define production readiness criteria for alpha, beta, and GA features.
+* Define a production readiness review gate and process for all features.
+* Utilize existing tooling with prow to enforce the process.
+
+### Non-Goals
+
+* Building new tooling to enforce the process.
+
+## Proposal
+
+* Document production readiness criteria in a process document in the
+  kubernetes/community repository. Different levels of readiness may be
+  specified for different feature maturity levels.
+
+* Develop a production readiness questionnaire to ensure that the feature
+  authors consider and document operational aspects of the feature. Example
+  items:
+  * Feature enablement
+    - How can this feature be disabled in a live cluster? Will that require
+      downtime for the control plane?
+  * Dependencies
+    - Does this feature depend on any specific services running in the cluster
+      (e.g., a metrics service)?
+    - How does this feature respond to complete failures of the services on
+      which it depends?
+    - How does this feature respond to degraded performance or high error rates
+      from services on which it depends?
+  * Monitoring requirements
+    - How can an operator determine if the feature is in use by workloads?
+    - How can an operator determine if the feature is functioning properly?
+    - What are the service level indicators an operator can use to determine the
+      health of the service?
+    - What are reasonable service level objectives for the feature?
+  * Debugging
+  * etc.
+
+* Establish a production readiness review team, label, and CI check to prevent
+  the merging of feature promotion PRs that lack production readiness.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate.
+Think broadly.
+For example, consider both security and how this will impact the larger kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.
+Anything that would count as tricky in the implementation and anything particularly challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage expectations).
+Please adhere to the [Kubernetes testing guidelines][testing-guidelines] when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+
+### Graduation Criteria
```

Comment:
does this section roll into the main kep document for a feature which also calls out graduation criteria?

Reply:
I think there will be something new in the main KEP template as part of implementing this KEP.